google/cloud/spanner/internal/session_pool.cc

```diff
@@ -197,6 +204,48 @@ void SessionPool::Erase(std::string const& session_name) {
   }
 }
 
+Status SessionPool::CreateMultiplexedSession() {
+  std::unique_lock<std::mutex> lk(mu_);
+  if (!multiplexed_session_ || multiplexed_session_->is_bad()) {
```

Comment:
nit: it seems we could return early, consider:

```suggestion
  if (multiplexed_session_ && !multiplexed_session_->is_bad()) return Status{};
```

Though I guess that would make reacquiring the lock slightly uglier.

Reply:
Left as is.